<launch>
    <arg name="robot_width" default="0.697"/>
    <arg name="wheel_radius" default="0.1825"/>
    <arg name="kinematics_type" default="differential"/>
    <arg name="kalman_params" default="false"/>

    <arg name="robot_length" value="0.44"/>

    <group if="$(arg kalman_params)">
        <node pkg="panther_driver" name="panther_driver" type="driver_node.py" output="screen" required="true">
            <param name="can_interface" type="string" value="panther_can"/>
            <param name="kinematics_type" type="string" value="$(arg kinematics_type)"/>
            
            <param name="publish_tf" type="bool" value="false"/>
            <param name="publish_pose" type="bool" value="false"/>
            <param name="publish_odometry" type="bool" value="true"/>

            <param name="odom_frame" type="string" value="odom"/>
            <param name="base_link_frame" type="string" value="base_link"/>

            <param name="robot_width" type="double" value="$(arg robot_width)"/>
            <param name="robot_length" type="double" value="$(arg robot_length)"/>
            <param name="wheel_radius" type="double" value="$(arg wheel_radius)"/>

            <param name="eds_file" type="string" value="$(find panther_driver)/params/roboteq_motor_controllers_v60.eds"/>
            <param name="measurements_file" type="string" value="$(find panther_driver)/params/measurements.yaml"/>
        </node>  
    </group>

    <group unless="$(arg kalman_params)">
        <node pkg="panther_driver" name="panther_driver" type="driver_node.py" output="screen" required="true">
            <param name="can_interface" type="string" value="panther_can"/>
            <param name="kinematics_type" type="string" value="$(arg kinematics_type)"/>
            
            <param name="publish_tf" type="bool" value="true"/>
            <param name="publish_pose" type="bool" value="true"/>
            <param name="publish_odometry" type="bool" value="false"/>

            <param name="odom_frame" type="string" value="odom"/>
            <param name="base_link_frame" type="string" value="base_link"/>
            
            <param name="robot_width" type="double" value="$(arg robot_width)"/>
            <param name="robot_length" type="double" value="$(arg robot_length)"/>
            <param name="wheel_radius" type="double" value="$(arg wheel_radius)"/>
            
            <param name="eds_file" type="string" value="$(find panther_driver)/params/roboteq_motor_controllers_v60.eds"/>
            <param name="measurements_file" type="string" value="$(find panther_driver)/params/measurements.yaml"/>
        </node>    
    </group> 
</launch>
